### PR TITLE
Fix pyenv issue for macOS

### PR DESCRIPTION
**Pull request by dilipped**
See b/187701234#comment12 for context.

### CODE REVIEW

File: test/ci/kokoro/run_integ_tests.sh [hunk 1]
```diff
@@ -54,18 +54,16 @@ function latest_python_release {
...
-    if [[ $KOKORO_JOB_NAME =~ "macos" ]]; then
-      brew update
-      brew install pyenv
-      eval "$(pyenv init -)"
-    fi
+    # MacOS VM does not have pyenv installed by default.
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    export PYENV_ROOT="$HOME/.pyenv"
+    export PATH="$PYENV_ROOT/bin:$PATH"
+    eval "$(pyenv init --path)"
```
Comment (NickGoog) [inline test/ci/kokoro/run_integ_tests.sh:61]: Why not just run `pyenv init --path`?

**Review by NickGoog — APPROVED** [review]

**Comment by NickGoog** [comment]
Make sure this isn't your invocation: http://sponge2/50a86890-70c3-4502-9dd0-ba6184299c14

File: test/ci/kokoro/run_integ_tests.sh (hunk 1 above)
Comment (dilipped) [reply test/ci/kokoro/run_integ_tests.sh:61]: The way that command works is that it will output a bunch of other commands, and we want to execute those.

**Comment by dilipped** [comment]
> Make sure this isn't your invocation: http://sponge2/50a86890-70c3-4502-9dd0-ba6184299c14

That one is from continous build, so it is not using this code.

**Pull request merged**